packages/core/src/render3/view_manipulation.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Injector} from '../di/injector';
+import {DehydratedContainerView} from '../hydration/interfaces';
+import {assertDefined} from '../util/assert';
+
+import {assertLContainer, assertLView} from './assert';
+import {renderView} from './instructions/render';
+import {createLView} from './instructions/shared';
+import {CONTAINER_HEADER_OFFSET, LContainer, NATIVE} from './interfaces/container';
+import {TContainerNode} from './interfaces/node';
+import {RComment, RElement} from './interfaces/renderer_dom';
+import {DECLARATION_LCONTAINER, FLAGS, LView, LViewFlags, QUERIES, RENDERER, T_HOST, TVIEW} from './interfaces/view';
+import {addViewToDOM, destroyLView, detachView, getBeforeNodeForView, insertView, nativeParentNode} from './node_manipulation';
+import {getTNode} from './util/view_utils';
+
+export function createAndRenderEmbeddedLView<T>(
+    hostLView: LView<unknown>, templateTNodeIdx: number, context: T,
```

Comment:
@pkozlowski-opensource I'd propose that we pass TNode here (and call `getTNode` outside) to make this function more generic (so we can use to elsewhere in the code that it not designed to work with multiple template nodes):
```suggestion
    hostLView: LView<unknown>, embeddedViewTNode: TNode, context: T,
```

Additional context: I was integrating those helpers into the defer prototype and at certain points I have a TNode already, so it'd be helpful to just pass it through.

We can still have a similar function where we accept template index and this function can call `getTNode` and `createAndRenderEmbeddedLView` inside.

Reply:
Yeh, I had the version with TNode initially but then changed to index in order to avoid checks that a given node belongs to a declaration TView. But honestly I don't mind either way so if TNode version makes it easier to work with, let's go with it.

Changed.